### PR TITLE
[widgets/shortcuts] Add includeSemantics property to the Shortcuts widget.

### LINKED ISSUE #152076: Add includeSemantics key to Shortcuts
### Use case

Because Shortcuts is an unfocusable Focus widget, it inserts a semantics node into the semantics tree. This can interfere with testing. 

### Proposal

Give the option to remove semantics by exposing the FocusNode.includeSemantics property in the Shortcuts widget.

```dart
class _ShortcutsState extends State<Shortcuts> {
 // ...

@override
  Widget build(BuildContext context) {
    return Focus(
      debugLabel: '$Shortcuts',
      canRequestFocus: false,
      onKeyEvent: _handleOnKeyEvent,
      includeSemantics: widget.includeSemantics, // Add this key
      child: widget.child,
    );
  }
```

### PR DESCRIPTION
**Pull request by davidhicks980**
Adds an `includeSemantics` property to the Shortcuts widget.

@gspencergoog and I discussed this change to make introducing Shortcuts easier on the MenuAnchor widget. This change should also make testing semantics trees slightly less onerous (assuming this doesn't harm accessibility, that is).

On the point of accessibility, I'm not sure what useful semantic information is exposed by the Shortcuts widget's semantic node, since it's basically just an unfocusable keyboard listener. However, I kept the `includeSemantics` defaulting to true so as to not break semantics tests in the Flutter repo.

Fixes https://github.com/flutter/flutter/issues/152076

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md
[test-exempt]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md
[Features we expect every widget to implement]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/blob/main/docs/contributing/Chat.md
[Data Driven Fixes]: https://github.com/flutter/flutter/blob/main/docs/contributing/Data-driven-Fixes.md

### CODE REVIEW

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

File: packages/flutter/test/widgets/shortcuts_test.dart [hunk 1]
```diff
@@ -1255,6 +1257,50 @@ void main() {
       expect(invoked, 1);
       invoked = 0;
     });
+
+     testWidgets('Shortcuts does not insert a semantics node when includeSemantics is false', (WidgetTester tester) async {
```
Comment (justinmc) [inline packages/flutter/test/widgets/shortcuts_test.dart:1261]: I think the indentation might be over by 1 space here?

**Review by justinmc — APPROVED** [review]
LGTM :+1:

File: packages/flutter/test/widgets/shortcuts_test.dart (hunk 1 above)
Comment (davidhicks980) [reply packages/flutter/test/widgets/shortcuts_test.dart]: Removed! Thank you for catching that.

**Pull request merged**